Update Kotlin mocking dependencies to be able to run test on JDK 21

JorgeMucientes opened the pull request:
### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Currently mockito-kotlin uses net.bytebuddy:byte-buddy dependency which doesn't support JDK 21 (default embedded JDK in Android Studio Ladybug onwards). This commit updates this testing deps to use the latest version of byte-buddy:1.14.18 and be able to run unit tests from AS without hassle.

Without this updates running test from AS (when jbr-21 Jetbrains default runtime embedded in AS Ladybug onwards) will lead to the following error: 
```
Underlying exception : org.mockito.exceptions.base.MockitoException: Could not modify all classes [class java.lang.Object, class com.woocommerce.android.viewmodel.ResourceProvider] ...
```

Extra context can be found [here](https://github.com/mockito/mockito/issues/3328) and [here](https://github.com/mockito/mockito/issues/3321)

### Testing information
Just getting all PR checks green should be enough. But if you want to ensure your test run locally in later AS version see the below desc. 

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
Tested running unit tests locally from AS Meerkat with Jetpabrains `jbr-21` set as Gradle JDK: 

![Screenshot 2025-01-27 at 12 07 21](https://github.com/user-attachments/assets/962d6dd0-f5b8-4769-aec9-edbf21c689e6)


- [X] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

JorgeMucientes committed:
Update kotlin mocking dependencies to be able to run test on JDK 21

Currently mockito-kotlin uses net.bytebuddy:byte-buddy dependency which doesn't support JDK 21 (default embedded JDK in Android Studio Ladybug onwards). This commit updates this testing deps to use the latest version of byte-buddy:1.14.18 and be able to run unit tests from AS without hassle.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes commented on gradle/libs.versions.toml:
```diff
@@ -38,7 +38,7 @@ apache-http-client-android = '4.3.5.1'
 automattic-measure-builds = '2.1.2'
 automattic-tracks = '5.0.0'
-assertj = '3.24.1'
+assertj = '3.26.3'
```
Asserj also uses byte-buddy so updated it to latest version as well.

irfano replied:
The latest version is `3.27.3`.

wzieba commented:
👋 Thanks for addressing it! I've experience this when working on vendoring FluxC. It's possible to remove `mockito-inline` as this is now bundled with core mockito, like this: https://github.com/woocommerce/woocommerce-android/pull/13304/commits/4c3a665bc4f1ee39e387ed6044c84be731a3414f . But it's completely fine if you decide to merge this as it is too, I can remove the `inline` with the FluxC vendor PR.

irfano reviewed (commented):
I reviewed the release notes for the new library updates and didn’t find anything relevant to us. Since these are only test libraries, a green check in the CI pipeline should suffice.
Regarding the assertj version, there is a newer version, `3.27.3`. I tested it and didn’t encounter any issues that would prevent us from upgrading. Is there any reason not tp upgrade it?

JorgeMucientes committed:
Update assertj to latest version

JorgeMucientes committed:
Remove Mockito inline

mockito-inline is now bundled with core mockito so its not needed anymore

JorgeMucientes commented:
Thanks for the input folks 🙏🏼 

- I've removed `mockito-inline` dependency @wzieba 👍🏼 
- Updated assertj to latest version @irfano 👍🏼. The reason I had updated it to `3.26.3` is simply because when tapping on AS  "alt-enter" in the `libs.version.toml` file to update assertj, that is the latest version it upgraded it to. Don't know why it didn't update it to the latest existing version.

irfano reviewed (approved):
LGTM! 👍🏻
